nbconvert/nbconvertapp.py

```diff
@@ -204,13 +238,23 @@ def _postprocessor_class_changed(self, name, old, new):
             self.postprocessor_factory = import_item(new)
 
 
-    # Other configurable variables
-    export_format = CaselessStrEnum(get_export_names(),
-        default_value="html",
+    export_format = Unicode(
+        values='html',
+        allow_none=False,
         config=True,
-        help="""The export format to be used."""
+        help="""The export format to be used, either one of the built-in formats, or a dotted object name that represents the import path for an `Exporter` class"""
     )
 
+    def _export_format_changed(self, old, new):
+        try:
+            _validate_exporter(new)
+        except ValueError as e:
+            sys.exit(e.args)
+        except ImportError as e:
+            self.log.debug(e, exc_info=True)
+            sys.exit("Cannot import exporter '%s'" % new)
```

Comment:
> Yuck, no, traitlet validation should never call sys.exit()

Yeah, I need to find a better way, but them that was to avoid traceback on bad exporter.



> Just let traitlets check that it's a string, and throw an error when we try to use it if it's the wrong string

I would prefer to check earlier, otherwise you end up with an exception relatively late in the process. But if it's really cleaner why not.